app/controllers/api/public/v1/application_controller.rb

```diff
@@ -11,12 +11,32 @@ def jsonapi_404
           render jsonapi: nil, status: :not_found
         end
 
+        def jsonapi_pagination(collection)
+          collection.present? && pagy_scope.present? ? pagination_links : {}
+        end
+
       private
 
-        def paginate(scope)
-          _pagy, paginated_records = pagy(scope, items: per_page, page: page)
+      # child must define pagy_scope method for paginations related items
+        def pagy_scope; end
```

Comment:
It's starting to feel like these pagy related methods should live in a separate module as there's a handful of them woven into our `ApplicationController`. These methods seem to just customise Pagy to work to our requirements.

Reply:
Done it in this commit https://github.com/DFE-Digital/teacher-training-api/pull/1690/commits/eb291d9ed80ae11ca31f9360a62623b5918194df